server/src/main/java/org/opensearch/search/startree/filter/provider/StarDateFieldMapper.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.startree.filter.provider;
+
+import org.opensearch.common.time.DateFormatter;
+import org.opensearch.common.time.DateMathParser;
+import org.opensearch.index.compositeindex.datacube.DateDimension;
+import org.opensearch.index.compositeindex.datacube.startree.index.StarTreeValues;
+import org.opensearch.index.compositeindex.datacube.startree.utils.date.DateTimeUnitRounding;
+import org.opensearch.index.mapper.CompositeDataCubeFieldType;
+import org.opensearch.index.mapper.DateFieldMapper.DateFieldType;
+import org.opensearch.index.mapper.MappedFieldType;
+import org.opensearch.index.query.RangeQueryBuilder;
+import org.opensearch.search.internal.SearchContext;
+import org.opensearch.search.startree.StarTreeQueryHelper;
+import org.opensearch.search.startree.filter.DimensionFilter;
+import org.opensearch.search.startree.filter.RangeMatchDimFilter;
+
+import java.time.ZoneId;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.LongSupplier;
+
+class StarDateFieldMapper implements DimensionFilterMapper {
+
+    CompositeDataCubeFieldType compositeDataCubeFieldType;
+    LongSupplier nowSupplier;
+    String subDimensionField;
+
+    public StarDateFieldMapper(SearchContext searchContext) {
+        this.nowSupplier = () -> searchContext.getQueryShardContext().nowInMillis();
+        this.compositeDataCubeFieldType = (CompositeDataCubeFieldType) searchContext.mapperService()
+            .getCompositeFieldTypes()
+            .iterator()
+            .next();
+    }
+
+    @Override
+    public DimensionFilter getExactMatchFilter(MappedFieldType mappedFieldType, List<Object> rawValues) {
+        // Finding exact match is not possible since star-tree stores the data for respective calendar intervals only
```

Comment:
Can we elaborate this comment ? @sandeshkr419 @bharath-techie

Reply:
I think we can evaluate the performance once for the cases you mentioned in later iterations based on the query weight and aggregation complexity.

I've mentioned this in date field parent issue for now: https://github.com/opensearch-project/OpenSearch/issues/17270

Also, added TODOs at relevant places.